File: nodes/quartiles/quartiles.process.py
async def process(inputs, settings, config, nodeConfig):
  numbers = inputs.get("numbers", [])
  if not isinstance(numbers, list):
    numbers = [numbers]
    
  # Validate all inputs are numbers
  for n in numbers:
    try:
      float(n)
    except (ValueError, TypeError):
      raise ValueError("Input contains non-numeric values that cannot be processed")
  
  # Convert all values to numbers
  valid_numbers = [float(n) for n in numbers]
  
  if not valid_numbers:
    return {
      "q1": 0,
      "q2": 0,
      "q3": 0,
      "iqr": 0
    }
    
  sorted_nums = sorted(valid_numbers)
  length = len(sorted_nums)
  
  # Handle single value case
  if length == 1:
    return {
      "q1": sorted_nums[0],
      "q2": sorted_nums[0],
      "q3": sorted_nums[0],
      "iqr": 0
    }
  
  def get_median(arr):
    if not arr:
      return 0
    n = len(arr)
    mid = n // 2
    return arr[mid] if n % 2 else (arr[mid - 1] + arr[mid]) / 2
  
  mid = length // 2
  q2 = get_median(sorted_nums)
  
  # Calculate Q1 (median of lower half)
  lower_half = sorted_nums[:mid]
  q1 = get_median(lower_half)
  
  # Calculate Q3 (median of upper half)
  upper_half = sorted_nums[mid + 1:] if length % 2 else sorted_nums[mid:]
  q3 = get_median(upper_half)
  
  return {
    "q1": q1,
    "q2": q2,
    "q3": q3,
    "iqr": q3 - q1
  } 

File: sdks/nodejs/nodes/quartiles/quartiles.process.js
export default async ({inputs, settings, config}) => {

  const numbers = Array.isArray(inputs.numbers) ? inputs.numbers : [inputs.numbers];
  
  // Validate all inputs are numbers
  for (const n of numbers) {
    if (n === null || n === undefined || isNaN(Number(n))) {
      throw new Error("Input contains non-numeric values that cannot be processed");
    }
  }
  
  // Convert all values to numbers
  const validNumbers = numbers.map(n => Number(n));
  
  if (validNumbers.length === 0) {
    return {
      q1: 0,
      q2: 0,
      q3: 0,
      iqr: 0
    };
  }
  
  const sorted = [...validNumbers].sort((a, b) => a - b);
  
  // Handle single value case
  if (sorted.length === 1) {
    return {
      q1: sorted[0],
      q2: sorted[0],
      q3: sorted[0],
      iqr: 0
    };
  }
  
  // Helper function to get median of an array
  const getMedian = (arr) => {
    if (arr.length === 0) return 0;
    const mid = Math.floor(arr.length / 2);
    return arr.length % 2 === 0 
      ? (arr[mid - 1] + arr[mid]) / 2
      : arr[mid];
  };
  
  const mid = Math.floor(sorted.length / 2);
  const q2 = getMedian(sorted);
  
  // Calculate Q1 (median of lower half)
  const lowerHalf = sorted.slice(0, mid);
  const q1 = getMedian(lowerHalf);
  
  // Calculate Q3 (median of upper half)
  const upperHalf = sorted.slice(sorted.length % 2 ? mid + 1 : mid);
  const q3 = getMedian(upperHalf);
  
  return {
    q1,
    q2,
    q3,
    iqr: q3 - q1
  };
}; 